npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/registry_finder.rb

```diff
@@ -174,14 +174,14 @@ def npmrc_registries
 
           registries = []
           npmrc_file.content.scan(NPM_AUTH_TOKEN_REGEX) do
-            next if Regexp.last_match[:registry].include?("${")
+            next if Regexp.last_match&.[](:registry)&.include?("${")
```

Comment:
Unfortunately I can't use `fetch` here like I do in `Hash` i.e. `Regexp.last_match&.fetch(:registry)`. `last_match` returns [a `MatchData`][1], not a `Hash`.

[1]: https://docs.ruby-lang.org/en/3.3/MatchData.html